packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1342,6 +1344,37 @@ abstract class FlutterCommand extends Command<void> {
     );
   }
 
+  // This adds the Dart defines used to access various Flutter version information at runtime.
+  void _addFlutterVersionToDartDefines(List<String> dartDefines) {
+    const List<String> flutterVersionDartDefines = <String>[
+      'FLUTTER_VERSION',
+      'FLUTTER_CHANNEL',
+      'FLUTTER_GIT_URL',
+      'FLUTTER_FRAMEWORK_REVISION',
+      'FLUTTER_ENGINE_REVISION',
+      'FLUTTER_DART_VERSION',
+    ];
+
+    for (final String dartDefine in flutterVersionDartDefines) {
+      if (globals.platform.environment[dartDefine] != null) {
+        throwToolExit('$dartDefine is used by the framework and cannot be set in the environment. '
+          'Use FlutterVersion to access it in Flutter code');
+      }
+      if (dartDefines.any((String define) => define.startsWith(dartDefine))) {
+        throwToolExit('$dartDefine is used by the framework and cannot be '
+          'set using --${FlutterOptions.kDartDefinesOption} or --${FlutterOptions.kDartDefineFromFileOption}. '
+          'Use FlutterVersion to access it in Flutter code');
+      }
+    }
+
+    dartDefines.add('FLUTTER_VERSION=${globals.flutterVersion.frameworkVersion}');
```

Comment:
we are trying to avoid using `globals` in new code; can you pass the flutterVersion object in as a dependency of this one instead?

Reply:
Done